File: backend/src/main/java/com/project/dto/assessment/request/AssessmentSubmitRequestDTO.java
package com.project.dto.assessment.request;

import java.util.List;

import lombok.Data;

@Data
public class AssessmentSubmitRequestDTO {

    private List<AnswerDTO> answers;

    @Data
    public static class AnswerDTO {
        private Long questionId;
        private Integer value;
    }
}

File: backend_dotnet/SafeMind.Core/DTOs/Assessment/Request/AssessmentSubmitRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace SafeMind.Core.DTOs.Assessment.Request
{
    public class AssessmentSubmitRequestDTO
    {
        [Required]
        public List<AnswerDTO> Answers { get; set; } = new List<AnswerDTO>();

        public class AnswerDTO
        {
            [Required]
            public long QuestionId { get; set; }
            
            [Required]
            public int Value { get; set; }
        }
    }
}
